### PR TITLE
Implements new indicator VWMA - Volume Weighted Moving Average

### LINKED ISSUE #8128: Implement Volume Weighted Moving Average
#### Expected Behavior
Supports Volume Weighted Moving Average (see [talipp implementation](https://github.com/nardew/talipp/blob/main/talipp/indicators/VWMA.py#L9)).

Refer to [LEAN Engine > Contributions > Indicators](https://www.quantconnect.com/docs/v2/lean-engine/contributions/indicators).

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by femtotrader**
#### Description

Implement new indicator - VWMA (Volume Weighted Moving Average)

#### Related Issue
Closes #8128

#### Motivation and Context
The VWMA indicator wasn't implemented.
(not to be  confused with VWAP)

#### Requires Documentation Change

Doc for this indicator should be created but some docstring are written

#### How Has This Been Tested?

comparison with talipp computation of VWMA(20) of SPY candle data using [generate_reference_data_from_talipp.py](https://github.com/QuantConnect/Lean/blob/master/Tests/TestData/generate_reference_data_from_talipp.py)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by femtotrader** [comment]
The followings tests are failing
- `AcceptsRenkoBarsAsInput`
- `WarmsUpProperly`
but others are passing.

Any idea?

**Comment by femtotrader** [comment]
The followings test is failing
- `AcceptsRenkoBarsAsInput`
but others are passing.

Any idea?

**Comment by AlexCatarino** [comment]
Since this is a moving average, we need to add it to the MovingAverageType enum after https://github.com/QuantConnect/Lean/pull/8177 is merged.

**Comment by femtotrader** [comment]
Will also add

            indicator = MovingAverageType.VWMA.AsIndicator(name, 9);
            Assert.IsInstanceOf(typeof(VolumeWeightedMovingAverage), indicator);

to

    MovingAverageTypeExtensionsTests.cs

**Comment by femtotrader** [comment]
I wonder how I can add VWMA to MovingAverageTypeExtensions.

This kind of MA is a bit different from others because it takes a bar as input... not a price.

Adding 


                case MovingAverageType.VWMA:
                    return new VolumeWeightedMovingAverage(period);

to MovingAverageTypeExtensions.cs ... next to others MA

and I'm getting a compile error

    Severity	Code	Description	Project	File	Line	Suppression State	Details
    Error (active)	CS0029	Cannot implicitly convert type 'QuantConnect.Indicators.VolumeWeightedMovingAverage' to 'QuantConnect.Indicators.IndicatorBase<QuantConnect.Indicators.IndicatorDataPoint>'	QuantConnect.Indicators	C:\Users\Admin\git\github\femto\Lean\Indicators\MovingAverageTypeExtensions.cs	76

File: Indicators/VolumeWeightedMovingAverage.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the Volume Weighted Moving Average (VWMA)
+    /// </summary>
+    public class VolumeWeightedMovingAverage : TradeBarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+
+        private IndicatorBase<IndicatorDataPoint> _rollingSumS { get; }
+        private IndicatorBase<IndicatorDataPoint> _rollingSumV { get; }
```
Comment (Martin-Molinero) [inline Indicators/VolumeWeightedMovingAverage.cs:27]: Smees this PR is missing to address previous reviews, 
- XML description improvement
- privates should be fields, no need to be properties
- `..SumS/..SumV` should be expanded to their complete names
- please push `IsReady` above constructor
- variable `_sumV` in method scope name is off codying style
- missing brackets
-

File: Algorithm/QCAlgorithm.Indicators.cs [hunk 2]
```diff
@@ -2242,6 +2242,24 @@ public IntradayVwap VWAP(Symbol symbol)
...
 
+        /// <summary>
+        /// Creates a new VolumeWeightedMovingAverage indicator for the symbol. The indicator will be automatically
+        /// updated on the given resolution.
+        /// </summary>
+        /// <param name="symbol">The symbol whose VWMA we want</param>
+        /// <param name="period">The smoothing period used to smooth the computed VWMA values</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to casting the input value to a TradeBar</param>
+        /// <returns>A new AverageTrueRange indicator with the specified smoothing type and period</returns>
```
Comment (Martin-Molinero) [inline Algorithm/QCAlgorithm.Indicators.cs:2253]: - XML documentation seems off, copy/paste
- `InitializeIndicator` needs to receive the symbol too, please take a look at how it works 👍

**Review by Martin-Molinero — CHANGES_REQUESTED** [review]
Hey @femtotrader! Cool!
Smees this PR needs to address similar previous reviews, posting. Please review all PRs and address these reviews. At this point should focus 💯 on getting existing PRs ready, not opening new PRs for now 👍

**Review by Martin-Molinero — APPROVED** [review]
Thank you 👍

**Pull request merged**